### PR TITLE
HLRC: add support for the clear realm cache API

### PR DESCRIPTION
**Pull request by jaymode**
This change adds support for clearing the cache of a realm. The realms
cache may contain a stale set of credentials or incorrect role
assignment, which can be corrected by clearing the cache of the entire
realm or just that of a specific user.

Relates #29827

### CODE REVIEW

**Comment by jaymode** [comment]
cc @elastic/es-security

**Review by albertzaharovits — APPROVED** [review]
LGTM 👍 
If I were to add anything, I would more thoroughly test the `ClearRealmCacheResponse`, especially that now`AbstractStreamableXContentTestCase` is a thing. But the response is simple enough for this to not be necessary.

**Comment by jaymode** [comment]
run sample packaging tests

File: client/rest-high-level/src/main/java/org/elasticsearch/client/SecurityClient.java [hunk 1]
```diff
@@ -239,7 +241,37 @@ public void authenticateAsync(RequestOptions options, ActionListener<Authenticat
...
 
     /**
-     * Clears the native roles cache for a set of roles.
+     * Clears the cache in one or more realms.
+     * See <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-clear-cache.html">
+     * the docs</a> for more.
+     *
+     * @param request the request with the realm names and usernames to clear the cache for
+     * @param options the request options (e.g. headers), use {@link RequestOptions#DEFAULT} if nothing needs to be customized
+     * @return the response from the enable user call
```
Comment (jkakavas) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/SecurityClient.java]: s/enable user/clear cache

**Review by jkakavas — APPROVED** [review]
LGTM

**Pull request merged**